src/settings/types.rs

```diff
@@ -44,34 +44,30 @@ impl FromStr for PythonVersion {
     }
 }
 
-#[derive(Debug, Clone, Hash)]
-pub enum FilePattern {
-    Simple(&'static str),
-    Complex(Pattern, Option<Pattern>),
-}
-
-impl FilePattern {
-    pub fn from_user(pattern: &str, project_root: Option<&PathBuf>) -> Result<Self> {
-        let path = Path::new(pattern);
-        let absolute_path = match project_root {
-            Some(project_root) => fs::normalize_path_to(path, project_root),
-            None => fs::normalize_path(path),
-        };
-
-        let absolute = Pattern::new(&absolute_path.to_string_lossy())?;
-        let basename = if pattern.contains(std::path::MAIN_SEPARATOR) {
-            None
+pub fn from_user(
+    pattern: &str,
+    project_root: Option<&PathBuf>,
+) -> std::result::Result<Glob, globset::Error> {
+    let path = Path::new(pattern);
+    let absolute_path = match project_root {
+        Some(project_root) => fs::normalize_path_to(path, project_root),
+        None => fs::normalize_path(path),
+    };
+    let pattern = Glob::new(&{
+        if pattern.contains(std::path::MAIN_SEPARATOR) {
+            absolute_path.to_string_lossy()
         } else {
-            Some(Pattern::new(pattern)?)
-        };
-
-        Ok(FilePattern::Complex(absolute, basename))
-    }
+            std::borrow::Cow::Owned(
+                "{".to_owned() + &absolute_path.to_string_lossy() + "," + pattern + "}",
+            )
```

Comment:
I think this merits a comment. Is this creating a pattern that's effectively a union of the absolute and base patterns?

Would it be clearer to instead return two Globs, one for each of those two patterns?

Reply:
2 Globs made much more sense to me, so I basically reimplemented `FilePattern::Complex` in cc32661 🤣